eclipse-theia/theia
# Support 'browser-only' Theia

![image](https://github.com/eclipse-theia/theia/assets/8998368/a7044401-2bf5-4da5-bf86-67c2f970a768)

Adds tooling, adapts the current code base and provides an example application for the new 'browser-only' Theia application target. This target generates a Theia frontend application which can run without a backend, transforming the Theia application to a static site.

Implements #12852

#### What it does
Adapts tooling to:
- support new 'browser-only' application target
- support new 'frontendOnly' and 'frontendOnlyPreload' module declaration
- use http-server when starting 'browser-only' applications

Replaces backend services with browser-only variants
- implementation of BrowserFS-based filesystem
- implementation of 'ServiceConnectionProvider' which timeouts for all requests, enabling loading all Theia packages

Adds a browser-only example application and api-samples showcasing the customization of the new BrowserFS-based filesystem.

#### How to test
Build and run the 'browser-only' Theia example.

You can also find a hosted variant [here](https://all--browseronly-theia.netlify.app/).

Note that you can build the 'browser-only' Theia even without building the Node native parts via:
```
yarn install --ignore-scripts
yarn browser-only prepare:no-native
yarn browser-only build
```

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
